cli/stream_command.go

```diff
@@ -1315,8 +1325,8 @@ func (c *streamCmd) copyAndEditStream(cfg api.StreamConfig, pc *fisk.ParseContex
 		cfg.Placement = nil
 	}
 
-	if len(c.sources) > 0 || c.mirror != "" {
-		return cfg, fmt.Errorf("cannot edit mirrors or sources using the CLI, use --config instead")
+	if len(c.sources) > 0 || c.mirror != "" || c.subjectTransformSource != "" || c.subjectTransformDest != "" {
+		return cfg, fmt.Errorf("cannot edit mirrors, sources or transforms using the CLI, use --config instead")
```

Comment:
We should be able to allow editing igest transform though using flags right?

Reply:
I had it at first but then I realized the problem is that there is no way to use the flags in edit to remove an existing subject transform so I made it that you have to use a config file like source and mirror, and even when I did allow it you must specify both source and dest when doing an edit because otherwise for example edit with just specifying a new dest but not specifying source does it mean remove the existing source or leave it there if there's one?

TLDR just use a config file to make those kinds of stream changes, anyways changing things like replicas or tags I that you may need to do by hand it's good to be able to do with just arguments but IMHO editing a JSON should be the preferred way to change a stream config (though you do have to know the JSON schema).

Reply:
I added some stuff to the cli library that you can now tell the difference between `--foo=""` and the option not given.  So then this can be made editable by flags. 

https://github.com/nats-io/natscli/blob/84138d64116854a7c954fbb0d9e451657d2d7107/cli/stream_command.go#L171

See the `IsSetByUser()` option there, you check the variable that is set in that way and then you know if the intention of the user is to remove/unset something by specifically setting it to empty.

I havnt yet made the change to do it everywhere it makes sense but have been using it for new things

Reply:
Done. Also added a `--no-transform` flag since republish had one already (and it makes for an easier to find way to remove the transform since most people won't realize it can be done by setting source and dest to "")